server/events/vcs/github_client.go

```diff
@@ -579,13 +579,21 @@ func (g *GithubClient) MergePull(pull models.PullRequest, pullOptions models.Pul
 	if err != nil {
 		return errors.Wrap(err, "fetching repo info")
 	}
+	protection, resp, err := g.client.Repositories.GetBranchProtection(context.Background(), *repo.Owner.Name, *repo.Name, pull.BaseBranch)
```

Comment:
If git branch protections aren't set, will this pass? Do we need to check for branch protections first before callign this function?

Reply:
According to the GitHub api docs it will return 404 if branch protection is not set for the requested branch